source/base/Memory.ooc

```diff
@@ -0,0 +1,85 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use ooc-base
+use math
+
+Memory: cover {
+	_pointer: UInt8*
+	_size: Int
+	pointer ::= this _pointer
+	size ::= this _size
+	init: func@ {
+		this init(null, 0)
+	}
+	init: func@ ~fromSize (size: Int) {
+		this init(gc_malloc(size), size)
+	}
+	init: func@ ~fromData (=_pointer, =_size)
+	free: func@ -> Bool {
+		result: Bool
+		if (result = this _pointer != null)
+			gc_free(this _pointer)
```

Comment:
That is just a name and as long as we don't change the compiler we will need to define it.